source/client-side-encryption/client-side-encryption.rst

```diff
@@ -385,8 +386,61 @@ to authenticate.
       key: byte[96] or String // The master key used to encrypt/decrypt data keys. May be passed as a base64 encoded string.
    }
 
+   kmip: {
+      endpoint: String
+   }
+
 See `Why are extraOptions and kmsProviders maps?`_
 
+Drivers MUST enable TLS for all KMS connections.
+
+kmip provider TLS options
+`````````````````````````
+
+Drivers MUST provide TLS options to configure TLS connections for the "kmip"
+provider.
+
+The TLS options SHOULD be consistent with the existing TLS options for MongoDB
+server TLS connections. The TLS options MUST enable setting a custom client
+certificate, equivalent to the `tlsCertificateKeyFile` URI option.
+
+Drivers SHOULD provide API that is consistent with configuring TLS options for
+MongoDB server TLS connections. New API to support the options MUST be
+independent of the KMS provider to permit future extension. The following is an
+example:
+
+.. code:: typescript
+
+   class AutoEncryptionOpts {
+      // setTLSOptions accepts a map of KMS provider names to TLSOptions.
+      // Throws an exception if the KMS provider name is not "kmip".
+      setTLSOptions (opts Map<String, TLSOptions>)
+   }
+
+   class ClientEncryptionOpts {
+      // setTLSOptions accepts a map of KMS provider names to TLSOptions.
+      // Throws an exception if the KMS provider name is not "kmip".
+      setTLSOptions (opts Map<String, TLSOptions>)
+   }
+
+Drivers MUST raise an error if the TLS options are set to disable TLS.
+The error MUST contain the message "TLS is required".
+
+Drivers MUST raise an error if TLS options are set for any KMS provider other
```

Comment:
I'm still not sure about this one.  I can see a use case for overriding the default SSLContext in Java, for any provider.

Reply:
What is the use case? If we do add AWS / Azure / GCP, I may need to come up with a way to test.

Reply:
Maybe they are using an old JVM whose default trust store doesn't include the CA certs that are required to connect (I don't know if this is common in practice).

Reply:
I will try adding a test for AWS / Azure / GCP with the mock KMS server.

Reply:
Added `KMS TLS Options Test` to test AWS / Azure / GCP / KMS.